nixos/modules/services/web-apps/healthchecks.nix

```diff
@@ -0,0 +1,184 @@
+{ config, lib, pkgs, buildEnv, ... }:
+
+with lib;
+
+let
+  cfg = config.services.healthchecks;
+  pkg = pkgs.healthchecks;
+  boolToPython = b: if b then "True" else "False";
+  environment = {
+    PYTHONPATH = pkg.pythonPath;
+    STATIC_ROOT = cfg.dataDir + "/static";
+    SECRET_KEY_FILE = cfg.secretKeyFile;
+    DB_NAME = "${cfg.dataDir}/healthchecks.sqlite";
+    ALLOWED_HOSTS = lib.concatStringsSep "," cfg.allowedHosts;
+    DEBUG = boolToPython cfg.debug;
+    REGISTRATION_OPEN = boolToPython cfg.registrationOpen;
+  } // cfg.extraConfig;
+
+  environmentFile = pkgs.writeText "healthchecks-environment" (lib.generators.toKeyValue {} environment);
+
+  healthchecksManageScript = with pkgs; (writeShellScriptBin "healthchecks" ''
+    sudo -u healthchecks sh <<EOF
+      export $(cat ${environmentFile} | xargs);
```

Comment:
Not sure if we need quoting here.

Reply:
I think it's good enough for this use-case. Quoting values written to the environmentFile would be a good practice, but I am not sure if this simple service module is the right place for it. I've been surprised that there didn't seem to be a function to write an environmentFile with quoted values already in nixpkgs, but maybe I just missed it? :thinking:

Reply:
Asking in https://matrix.to/#/#dev:nixos.org resulted in a recommendation for https://skarnet.org/software/s6/s6-envdir.html . I could add that one if you'd like, but as said I personally think the current iteration works well enough.

Reply:
whatever you like